### PR TITLE
Major update.

### PR DESCRIPTION
**Pull request by davidlehn**
- This branch has been out there, and in use, for some time.
- Removes old things, adds news things, updates most things.
- Works well with `bedrock-{webpack,vue,quasar,fontawesome}` updates.
- Pardon the mess, but other `watch#` branches may need to stay around until projects are updated.

### CODE REVIEW

**Comment by davidlehn** [comment]
- There may be some more code that can be removed here.  The package.json bedrock manifest code is still processed.  It was useful for systemjs processing, but may not be needed at all anymore?
- The custom bedrockDependencies was also for systemjs.  It *could* still be used to override reading all deps with a custom list, but I'm unsure if that has a real use case?
- The above can be handled later as needed.

**Comment by dlongley** [comment]
@gannan08, @aljones15, @mattcollier -- please specifically weigh in on whether you think we still need the old features @davidlehn mentioned. Ideally we'd clean out the cruft and have less to maintain.

**Comment by mattcollier** [comment]
very much in favor of eliminating browserDependencies if possible.

**Comment by gannan08** [comment]
> * The above can be handled later as needed.

@davidlehn If this is the case then remove old code.

**Comment by davidlehn** [comment]
@mattcollier The default behavior is now to work as if `"browserDependencies": "all"` was specified, and it just reads all the dependencies.  I'm unsure if there's a use case where you would want to override that to only look at specific deps.  I think all our current projects should/need to be updated to just remove their `browserDependencies` configuration.

**Comment by davidlehn** [comment]
@gannan08 You are suggesting to just remove all the old stuff now?  I had been thinking just leaving some of the old support in there for the next release, update all the active projects, then we'll have more info to determine if we can clean up unused features even more.

**Comment by davidlehn** [comment]
`bedrock.{manifest,browserDependencies}` support now removed.

File: lib/config.js [hunk 1]
```diff
@@ -298,17 +84,6 @@ config.views.browserVersions = {
 };
 
 config.views.vars = {
```
Comment (dlongley) [inline lib/config.js]: Can we get rid of all these view vars as well? And remove the "data" thing we pass to the frontend?

File: views/layout.html [hunk 2]
```diff
@@ -26,18 +32,15 @@
...
       {% endif %}
     {% endif %}
   </head>
 
   {% if hasSupportedUserAgent() %}
-  <body class="{{(app.ngClass.body && '') || 'br-body'}}"
-    ng-class="app.ngClass.body"
-    ng-style="app.ngStyle.body">
+  <body class="br-body">
     {% if pageLayout != "error" %}
```
Comment (dlongley) [inline views/layout.html]: Are we still using `pageLayout`? Can we remove?

File: views/layout.html [hunk 3]
```diff
@@ -26,18 +32,15 @@
...
       {% include "data.html" %}
```
Comment (dlongley) [inline views/layout.html]: Can we remove `data.html`?

File: views/layout.html (hunk 2 above)
Comment (dlongley) [reply views/layout.html]: Are we still using `pageTitle`? Can we remove?

Comment (dlongley) [reply views/layout.html]: Can we simplify `layout.html` to just be `main.html` now?

Comment (dlongley) [reply views/layout.html]: Can we remove `br-app-load*` stuff?

File: lib/config.js (hunk 1 above)
Comment (mattcollier) [reply lib/config.js]: +1 to that!  Which I believe would eliminate that weird getViewsVars thing that links bedrock-rest to this module requiring bedrock-view and all it's deps to be installed wherever bedrock-rest is used.

**Comment by davidlehn** [comment]
Found an annoyance.  bedrock-webpack uses a frontend polyfill file that *may* require you to have top-level deps on core-js and regenrator-runtime.  It just depends on what other dependencies you have and how npm lays them out.  I added an upgrade note here about it.  We can probably ignore it for now unless someone wants to look into the issue and find a better solution.

**Review by dlongley — APPROVED** [review]
Approving -- my understanding is that another PR will follow on (prior to the new release) that removes the other things commented on here.

File: lib/config.js (hunk 1 above)
Comment (davidlehn) [reply lib/config.js]: Removed.  Will make PRs for other packages that I know use this.

**Pull request merged**